notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+
+## Vector Space Review
+- Definition of a vector
+- Scaling vectors
+- Adding vectors
+- Vector notations
+- The $\vec{1}$ vector
+
+- Span of a set of vectors
+- Vector space
+- Vector subspace
+
+
+- Angles between vectors
+- Vector length
+- Distance between of vectors
+- Orthogonal vectors (needs work)
+- Orthogonal vector spaces (omitted for now)
+- Projections of vectors
+
+#### Vector Definition
+A vector is defined by a length and a direction.
+
+<img src="../../notebooks-images/vector_space_review__vectors.png" width="500" />
+
+Notice that $\vec{x}$ and $\vec{y}$ have the same length and direction. They are equal vectors.
+
+#### Scaling and Adding Vectors
+To scale a vector is to change it's length.
+
+<img src="../../notebooks-images/vector_space_review__scaling.png" width="500" />
+
+Notice that $\vec{2x}$ and $\vec{y}$ have the direction but different lengths. They are not equal.
+
+To add two vectors $\vec{y} + \vec{z}$, take one step according to the length of $\vec{y}$, then immediately take one step according to the length of $\vec{z}$ (or vice versa). This is also known as triangle method, where you place the initial point of a vector on the terminal point of the other.
+
+<img src="../../notebooks-images/vector_space_review__adding.png" width="300" />
+
+#### Vector Notations
+Vectors are usually represented as Cartesian Coordinates.
+
+<img src="../../notebooks-images/vector_space_review__notation.png" width="500" />
+
+$$ \vec{x} = \begin{bmatrix} 1 \\ 4  \end{bmatrix} , \quad 
+   \vec{y} = \begin{bmatrix} 3 \\ 2  \end{bmatrix} , \quad 
+   \vec{z} = \begin{bmatrix} 4 \\ 0  \end{bmatrix}$$
+   
+In this notation, arithmetic operations we saw earlier become quite easy.
+
+$$ \vec{2x} = \begin{bmatrix} 2 \\ 8  \end{bmatrix} , \quad
+   \vec{-0.5z} = \begin{bmatrix} -2 \\ 0  \end{bmatrix} , \quad
+   \vec{2x + -0.5z} = \begin{bmatrix} 0 \\ 8  \end{bmatrix}
+$$
+
+<img src="../../notebooks-images/vector_space_review__notation_arithmetic.png" width="500" />
+
+
+Vectors can be added and scaled element-wise:
+
+$$a \vec{x} + b \vec{y} = \begin{bmatrix} a \ x_1 & + & b \  y_1 \\
+                                            & \vdots & \\
+                                          a \  x_n & + & b \  y_n
+                          \end{bmatrix}
+$$
+
+#### The $\vec{1}$ Vector
+In any $d$ dimensional space, $\vec{1}$ is the vector of all $1$'s: 
+$ \begin{bmatrix} 1 \\ \vdots \\ 1  \end{bmatrix}$
+
+#### Span
+The span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$ is the set of all possible linear combinations. For these $p$ vectors:
+
+$$ \{ c_1 \ \vec{v_1} + c_2 \ \vec{v_2} + \dots + c_p \ \vec{v_p} \ : \ \forall c_i \in F\}$$
+
+where $F$ is the field of the vector space (out of scope).
+
+#### Vector Spaces
+A vector space $V$ is the span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$, where each $\vec{v_i}$ is a $n \times 1$ dimensional column vector.
+
+#### Vector Subspaces
```

Comment:
Do we need this for this section? :OOOOO

Reply:
haha yea so Sam and I are thinking about having a linear algebra review section. we might move all the review to a different notebook